src/main/java/build/buildfarm/worker/CASFileCache.java

```diff
@@ -1508,6 +1508,30 @@ public synchronized void decrementReferences(
     decrementReferencesSynchronized(inputFiles, inputDirectories);
   }
 
+  /**
+   * The different contexts in which cache files are written to disk. Files are made non-writable in
+   * order to prevent operations from deleting them.
+   */
+  private void putDirectoryHardLink(Path newHardLink, Path path) throws IOException {
+    Files.createLink(newHardLink, path);
+    new File(newHardLink.toString()).setWritable(false, false);
+  }
+
+  private void putDirectoryFile(Path path, FileNode fileNode) throws IOException {
+    Files.createFile(path);
```

Comment:
This is an empty file creation? I would name it as such, and only pass the boolean executable